File: Java/Program/Handphone.java
// kelas Handphone yang merupakan subclass dari Sumisang
public class Handphone extends Sumisang {
    // atribut tambahan untuk Handphone
    private String androidVer;
    private String uiVer;
    private String processor;

    // konstruktor default
    public Handphone() {
        super();
    }

    // konstruktor parameterized
    public Handphone(String code, String name, int price, String androidVer, String uiVer, String processor) {
        super(code, name, price);
        this.androidVer = androidVer;
        this.uiVer = uiVer;
        this.processor = processor;
    }

    // setter dan getter untuk androidVer
    public String getAndroidVer() {
        return androidVer;
    }

    public void setAndroidVer(String androidVer) {
        this.androidVer = androidVer;
    }

    // setter dan getter untuk uiVer
    public String getUiVer() {
        return uiVer;
    }

    public void setUiVer(String uiVer) {
        this.uiVer = uiVer;
    }

    // setter dan getter untuk processor
    public String getProcessor() {
        return processor;
    }

    public void setProcessor(String processor) {
        this.processor = processor;
    }
}

File: CPP/Program/Handphone.cpp
#include <iostream>
#include <string>
#include "Sumisang.cpp"

using namespace std;

// Kelas Handphone, turunan dari Sumisang
class Handphone : public Sumisang{
    private:
        string androidVer;
        string uiVer;
        string processor;

    public:
        // Konstruktor default dan parameterized
        Handphone(){
        }

        Handphone(string code, string name, int price, string androidVer, string uiVer, string processor) : Sumisang(code, name, price){
            this->androidVer = androidVer;
            this->uiVer = uiVer;
            this->processor = processor;
        }

        // Setter dan Getter untuk androidVer
        void setAndroidVer(string androidVer){
            this->androidVer = androidVer;
        }

        string getAndroidVer(){
            return this->androidVer;
        }

        // Setter dan Getter untuk uiVer
        void setUiVer(string uiVer){
            this->uiVer = uiVer;
        }

        string getUiVer(){
            return this->uiVer;
        }

        // Setter dan Getter untuk processor
        void setProcessor(string processor){
            this->processor = processor;
        }

        string getProcessor(){
            return this->processor;
        }

        ~Handphone(){
        }
};